sdk/dotnet/Pulumi.Automation/StackSettings.cs

```diff
@@ -9,18 +9,24 @@ public class StackSettings
         /// <summary>
         /// This stack's secrets provider.
         /// </summary>
+        [YamlDotNet.Serialization.YamlMember(Alias="secretsprovider")]
+        [System.Text.Json.Serialization.JsonPropertyName("secretsprovider")]
```

Comment:
The code might be using a custom serializer, see StackSettingsConfigValueJsonConverter it looks very suspect. A test would make sure this is taking effect.